code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -86,9 +89,29 @@
 /datum/reagent/proc/affect_touch(var/mob/living/carbon/M, var/alien, var/removed)
 	return
 
-/datum/reagent/proc/overdose(var/mob/living/carbon/M, var/alien) // Overdose effect. Doesn't happen instantly.
-	M.adjustToxLoss(REM)
-	return
+/datum/reagent/proc/overdose(var/mob/living/carbon/M, var/alien)
+//	M.overdosing -= REM
+	if(od_type == "heart")
+		if (istype(M, /mob/living/carbon/human))
+			var/mob/living/carbon/human/C = M
+			var/obj/item/organ/internal/heart/H = C.internal_organs_by_name[O_HEART]
+			playsound(M.loc, 'sound/effects/singlebeat.ogg', 5, -1)	//Don't know how to keep this mob specific
+			C.pulse += 2

```

Comment:
You're adding 2 to the pulse every call of overdose(), which makes no sense simply for pulse using flags. Depending on where overdose() is called in the code, it will either break the value or be overwritten by handle_pulse().